File: backend/api-gateway/src/main/java/ru/animaltracker/apigateway/service/UserServiceImpl.java
package ru.animaltracker.apigateway.service;

import ru.animaltracker.apigateway.entity.User;
import ru.animaltracker.apigateway.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Mono;

@Slf4j
@Service
@RequiredArgsConstructor
public class UserServiceImpl implements UserService {
    private final UserRepository userRepository;

    @Override
    public Mono<User> getByUsername(String username) {return userRepository.findByUsername(username);}

    @Override
    public Mono<User> getById(Long id) {
        return userRepository.findById(id);
    }

    @Override
    public  Mono<User> getByEmail(String email) {return userRepository.findByEmail(email);}
}

File: backend/user-service/src/main/kotlin/ru/doedating/userservice/service/impl/UserServiceImpl.kt
package ru.doedating.userservice.service.impl

import ru.doedating.userservice.entity.User
import ru.doedating.userservice.repository.UserRepository
import ru.doedating.userservice.service.interfaces.UserService

class UserServiceImpl(
    private val userRepository: UserRepository,
) : UserService {
    override fun findByUsername(username: String): User? =
        userRepository.findByEmail(username)

    override fun findByEmail(email: String): User? =
        userRepository.findByEmail(email)
}